burn-core/src/optim/decay.rs

```diff
@@ -57,6 +57,16 @@ impl<B: Backend> WeightDecay<B> {
 
         (grad, WeightDecayState::new(grad_last_step))
     }
+
+    /// temp fix for Transform.
+    pub fn transform_temp_fix<const D: usize>(
+        &self,
+        grad: Tensor<B, D>,
+        tensor: Tensor<B, D>,
+    ) -> (Tensor<B, D>, Tensor<B, D>) {
+        let grad = tensor.clone().mul_scalar(self.penalty).add(grad);
+        (grad, tensor.clone())
```

Comment:
I think I miss read the PyTorch documentation when implementing the first version. `g_{t}` is actually calculated using `theta_{t-1}` not `theta_{t}`.

This is confusing since I expected the gradients of the forward pass to share the same timestep as the parameters that are used, but I don't make the rules 😅

So weight decay doesn't actually have a state, since the current tensor is available and that's why your new function works.

Reply:
yeah, that's the point here. Actually there's no state for `weight_decay`, for the `tensor` is just the state. 😄